src/sortStudents.ts

```diff
@@ -1,16 +1,56 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[]
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+function averageGrade(person: Student): number {
+  return person.grades
+    .reduce((total: number, cur: number) => total + cur, 0)
+    / person.grades.length;
+}
+
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  const copy: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Surname:
+    case SortType.Name:
+      return order === 'asc'
+        ? copy
+          .sort((x: Student, y: Student) => x[sortBy].localeCompare(y[sortBy]))
```

Comment:
types gonna be inferred (when you declare a variable `copy` with `Student[]` type), so no needs to write them directly
```suggestion
          .sort((x, y) => x[sortBy].localeCompare(y[sortBy]))
```